code/game/objects/effects/effect_system/effects_smoke.dm

```diff
@@ -16,61 +16,64 @@
 	layer = FLY_LAYER
 	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
 	var/steps = 0
-	var/lifetime = 5
+	var/lifetime = 10 SECONDS_TO_LIFE_CYCLES
 	var/direction
+	var/causes_coughing = FALSE
 
 /obj/effect/particle_effect/smoke/proc/fade_out(frames = 16)
-	if(alpha == 0) //Handle already transparent case
-		return
-	if(frames == 0)
-		frames = 1 //We will just assume that by 0 frames, the coder meant "during one frame".
-	var/step = alpha / frames
-	for(var/i = 0, i < frames, i++)
-		alpha -= step
-		if(alpha < 160)
-			set_opacity(0)
-		stoplag()
-
-/obj/effect/particle_effect/smoke/New()
+	animate(src, 2 SECONDS, alpha = 0, easing = EASE_IN | CIRCULAR_EASING)
```

Comment:
Nitpick, can you move this under the `Destroy()` proc?

Reply:
Not really, since kill_smoke() calls both this and QDEL_IN(2 SECONDS) so if I put this in Destroy() it'd need to delay itself to allow the animation to play out in full.